Adds an oscal directory and GitHub Actions workflow for upstream OSCAL content

jpower432 opened the pull request:
#### Description:

Adds initial vendor directory for OSCAL content with REV4 and REV5 NIST 800-53 catalogs and FedRAMP High profiles. Adds update-vendor.yaml for automated updates of content

#### Rationale:

Allow upstream OSCAL content to be updated, reviewed, and stored to aid to generating OSCAL component definitions.
Partially address issue #11106 

#### Review Hints:

This is part one to a two part feature add. The update-vendor.yaml was tested/verified locally using `act`.

Mab879 commented:
We store other vendor files (such as the STIG XML files) in `shared/references` that might be a better place for these files.

Mab879 commented on .github/workflows/update-vendor.yaml:
```diff
@@ -0,0 +1,55 @@
+          delete-branch: true
+          commit-message: "Update vendored OSCAL content"
+          add-paths: |
+            vendor/
```
Please add a new line at the end of the file.

Mab879 commented on .github/workflows/update-vendor.yaml:
```diff
@@ -0,0 +1,55 @@
+      - name: Update content
+        uses: peter-evans/create-pull-request@v5.0.2
+        with:
+          base: master
```
Please add a useful title for these PRs.

https://github.com/peter-evans/create-pull-request#action-inputs

Mab879 commented on .github/workflows/update-vendor.yaml:
```diff
@@ -0,0 +1,55 @@
+name: Update vendored OSCAL content
```
Per the project [style gude](https://complianceascode.readthedocs.io/en/latest/manual/developer/04_style_guide.html#yaml) please use 4 space indention.

Mab879 commented on .github/workflows/update-vendor.yaml:
```diff
@@ -0,0 +1,55 @@
+        uses: peter-evans/create-pull-request@v5.0.2
+        with:
+          base: master
+          branch: "vendor-update-${{ github.run_id }}"
```
A body on the PR might be useful as well.

Mab879 reviewed (changes requested):
Thanks for the PR.

I have a few minor points mainly around style.

jpower432 commented:
Thanks for the feedback @Mab879. I moved the directory to `shared/references/oscal`. Also changed some of the names to be more OSCAL specific since OSCAL is just one type of vendored file. Styling fixes also added in the last commit. Please let me know if any other changes are needed.

jpower432 commented:
Adding information on the last change. The workflow was originally pointing to the to the `oscal-content` v1.0.0 release, but there are newer version of the catalog. Added environment variables so the references are easier to change and updated to the latest commit on the repository before the update to OSCAL version 1.1.1. `compliance-trestle` does not support this version yet.

Mab879 commented on .github/workflows/update-oscal.yml:
```diff
@@ -0,0 +1,65 @@
+                      - usnistgov NIST 800-53 from "${{ matrix.variables.catalog-source }}"
+                      - GSA FedRAMP OSCAL profiles from "${{ matrix.variables.profile-source }}"
+                    
+                      Auto-generated by [create-pull-request](https://github.com/peter-evans/create-pull-request) GitHub action
```
Could you also provide the path to this GitHub Actions file as well?

Mab879 commented on shared/references/oscal/REAMDE.adoc:
Per the [style guide](https://complianceascode.readthedocs.io/en/latest/manual/developer/04_style_guide.html#asciidoc) please use Markdown unless there is a compelling reason.
